packages/interfaces/lib/scraper.ts

```diff
@@ -0,0 +1,52 @@
+import { IParser } from "./interfaces";
+import fetch from "node-fetch";
+
+/**
+ * A generalized scraper abstraction class
+ * This class can scrape different sites of pdfs
+ * @param IRet is the return interface for a scraped site or article
+ */
+export class Scraper<IRet> {
+  private urls: string[];
+  private parser: IParser<IRet>;
+  constructor(parser: IParser<IRet>, ...urls: string[]) {
+    this.urls = urls;
+  }
+
+
+  /**
+   * Retrieves the source code of a url
+   * @param url
```

Comment:
When you're using TS, you don't need to list params here (up to you).  The only time I would definitely add them to the jsdoc is if it's unclear what they're supposed to be